### PR TITLE
Safe notice on project page

### PR DESCRIPTION
**Pull request by johnnyd-eth**
**Only shows up when:**
1. The connected wallet is on the multisig.
2. This user has unsigned transactions in the queued list.
(the count in the tooltip only counts the unique nonces of transactions that the user hasn't signed)

Screen Shot 2022-10-13 at 10 32 30 pm

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/Project/ProjectHeader/GnosisSafeBadge.tsx [hunk 1]
```diff
@@ -21,17 +32,90 @@ export function GnosisSafeBadge({ href }: { href: string }) {
...
+        placement="bottom"
+        overlayStyle={{ width: 700 }}
+        title={
+          <>
+            <div>
+              <Trans>
+                {plural(unsignedNoncesOfUser.length, {
+                  one: '# transaction needs',
+                  other: '# transactions need',
+                })}{' '}
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/GnosisSafeBadge.tsx:97]: **praise** nice job here!

Remember that we want to translate full sentences though. I think we can add `your signature` to the end of each sentence here, and remove the `<Trans>` tag.

File: src/utils/safe.ts [hunk 2]
```diff
@@ -12,6 +12,41 @@ export function getUniqueNonces(transactions: SafeTransactionType[]) {
   ]
 }
 
+// returns subset of given transactions for which a given address has not signed
+export function getUnsignedTxsOfUser({
```
Comment (tomquirk) [inline src/utils/safe.ts:16]: **suggestion**

```suggestion
export function getUnsignedTxsForAddress({
```

File: src/utils/safe.ts [hunk 3]
```diff
@@ -12,6 +12,41 @@ export function getUniqueNonces(transactions: SafeTransactionType[]) {
...
+  const unsignedTxs: SafeTransactionType[] = []
+  transactions.forEach((transaction: SafeTransactionType) => {
+    let hasSigned = false
+    transaction.confirmations?.forEach((confirmation: GnosisSignature) => {
+      if (confirmation.owner.toLowerCase() === address) {
+        hasSigned = true
+      }
+    })
+    if (!hasSigned) unsignedTxs.push(transaction)
+  })
```
Comment (tomquirk) [inline src/utils/safe.ts:32]: Give this a crack (untested)

```suggestion
  const unsignedTxs: SafeTransactionType[] = transactions.filter(tx => {
  return tx.confirmations.some(
    confirmation => confirmation.owner.toLowerCase() === address,
  )
})

```

File: src/models/safe.ts [hunk 4]
```diff
@@ -5,6 +5,14 @@ export type GnosisSafe = {
   threshold: number
 }
 
+export type GnosisSignature = {
```
Comment (tomquirk) [inline src/models/safe.ts:8]: **praise**

File: src/components/Project/ProjectHeader/GnosisSafeBadge.tsx [hunk 5]
```diff
@@ -21,17 +32,90 @@ export function GnosisSafeBadge({ href }: { href: string }) {
...
-        <Trans>
-          This project is owned by a Safe.{' '}
-          <Link href={href}>See transactions</Link>.
-        </Trans>
-      }
-    >
-      <Icon component={SafeIcon} />
-    </Tooltip>
-  )
+  function DefaultBadge() {
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/GnosisSafeBadge.tsx:35]: **suggestion** its risky to define components within other components, can cause rendering issues. Can we move all these components outside, into their own scope?

File: src/components/Project/ProjectHeader/GnosisSafeBadge.tsx [hunk 6]
```diff
@@ -21,17 +32,90 @@ export function GnosisSafeBadge({ href }: { href: string }) {
...
+    )
+  }
+
+  // Red dot that appears conditionally next to Safe badge when connected user is on multisig and has unsiged tranctions
+  function Notice() {
+    return (
+      <div
+        style={{
+          position: 'absolute',
+          backgroundColor: 'red',
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/GnosisSafeBadge.tsx:57]: **suggestion** can we use whatever `red` color we have defined in our color pallete?

**Review by tomquirk — CHANGES_REQUESTED** [review]
COupla comments @johnnyd-eth but looking really good!

File: src/components/Project/ProjectHeader/GnosisSafeBadge.tsx (hunk 5 above)
Comment (johnnyd-eth) [reply src/components/Project/ProjectHeader/GnosisSafeBadge.tsx:35]: Good call, done

File: src/utils/safe.ts (hunk 3 above)
Comment (johnnyd-eth) [reply src/utils/safe.ts:32]: Very nice, just needed a `!` before the `.some` statement

**Comment by johnnyd-eth** [comment]
@tomquirk thanks for the comments ser! Addressed them all

**Review by tomquirk — APPROVED** [review]

**Pull request merged**